custom_components/energyscore/sensor.py

```diff
@@ -111,9 +111,19 @@ def __init__(self, hass, config):
             LAST_UPDATED: None,
         }
 
-        entity_reg = er.async_get(hass)
-        if entity_reg.async_get(self._price_entity).platform == "nordpool":
-            self._nordpool = True
+        self.hass = hass
+
+    @property
+    def is_nordpool(self) -> Optional[bool]:
+        if self._nordpool is None:
+            entity_reg = er.async_get(self.hass)
+            price_entity = entity_reg.async_get(self._price_entity)
+            if price_entity:
+                self._nordpool = price_entity.platform == "nordpool"
+            else:
+                # see https://github.com/knudsvik/EnergyScore/pull/52
+                _LOGGER.warn("Price entity is not defined, probably a misconfiguration or entity has no `unique_id`")
```

Comment:
Is it ok to use warn? This page says to use debug for "everything else": https://developers.home-assistant.io/docs/development_guidelines/?_highlight=_logger#use-new-style-string-formatting

Reply:
I must confess I don't understand the logic behind the idea of not adding warnings whenever necessary. Here we are in the case of user requiring to take some action to make the integration works.

Reply:
but let's be compliant :)